src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/validation/function.ts

```diff
@@ -6,930 +6,317 @@
-  if (!(isColumn(actualArg) || isIdentifier(actualArg)) || isParametrized(actualArg)) {
-    return messages;
+  /**
+   * Checks if the current license level is sufficient the given license requirement
+   */
+  private licenseOk(license: ESQLSignatureLicenseType | undefined): boolean {
+    const hasMinimumLicenseRequired = this.callbacks.hasMinimumLicenseRequired;
+    if (hasMinimumLicenseRequired && license) {
+      return hasMinimumLicenseRequired(license as ESQLLicenseType);
+    }
+    return true;
   }
 
-  const columnName = getQuotedColumnName(actualArg);
-  const columnExists = getColumnExists(actualArg, context);
-
-  if (parameterDefinition.constantOnly) {
-    messages.push(
-      getMessageFromId({
-        messageId: 'expectedConstant',
-        values: {
-          fn: astFunction.name,
-          given: columnName,
-        },
-        locations: actualArg.location,
-      })
-    );
-
-    return messages;
+  /**
+   * Checks if the function is available in the current context
+   */
+  private get availableHere(): boolean {
+    const { location } = getFunctionLocation(this.fn, this.parentCommand);
+    return this.definition?.locationsAvailable.includes(location) ?? false;
   }
 
-  if (!columnExists) {
-    messages.push(
-      getMessageFromId({
-        messageId: 'unknownColumn',
-        values: {
-          name: actualArg.name,
-        },
-        locations: actualArg.location,
-      })
-    );
-
-    return messages;
+  /**
+   * Checks if the function has a valid number of arguments
+   */
+  private get hasValidArity(): boolean {
+    const { min, max } = getMaxMinNumberOfParams(this.definition!);
+    const arity = this.fn.args.length;
+    return arity >= min && arity <= max;
   }
+}
 
-  if (actualArg.name === '*') {
-    // special case for COUNT(*)
-    return messages;
-  }
+export const PARAM_TYPES_THAT_SUPPORT_IMPLICIT_STRING_CASTING: FunctionParameterType[] = [
+  'date',
+  'date_nanos',
+  'date_period',
+  'time_duration',
+  'version',
+  'ip',
+  'boolean',
+];
 
-  if (!checkFunctionArgMatchesDefinition(actualArg, parameterDefinition, context, parentCommand)) {
-    const columnHit = getColumnForASTNode(actualArg, {
-      fields: context.fields,
-      userDefinedColumns: context.userDefinedColumns,
-    });
-    const isConflictType = columnHit && 'hasConflict' in columnHit && columnHit.hasConflict;
-    if (!isConflictType) {
-      messages.push(
-        getMessageFromId({
-          messageId: 'wrongArgumentType',
-          values: {
-            name: astFunction.name,
-            argType: parameterDefinition.type as string,
-            value: actualArg.name,
-            givenType: columnHit!.type,
-          },
-          locations: actualArg.location,
-        })
-      );
+/**
+ * Returns a signature matching the given types if it exists
+ * @param definition
+ * @param types
+ */
+function getSignaturesMatchingTypes(
+  definition: FunctionDefinition,
```

Comment:
NiT: I think we should pass only  signatures here, not all definition. The scope are signatures and also the name of the function suggest it.

Reply:
Done in https://github.com/elastic/kibana/pull/230139/commits/cfa683048f17ac8fe13b157b95cf349e5a6ede96